src/loader.ts

```diff
@@ -223,6 +211,25 @@ export async function loadConfig<
   return r;
 }
 
+export async function resolveConfigPath<
+  T extends UserInputConfig = UserInputConfig,
+  MT extends ConfigLayerMeta = ConfigLayerMeta,
+>(options: LoadConfigOptions<T, MT>): Promise<string | undefined> {
+  normalizeLoadOptions(options);
+
+  // Load dotenv
+  if (options.dotenv) {
+    await setupDotenv({
+      cwd: options.cwd,
+      ...(options.dotenv === true ? {} : options.dotenv),
+    });
+  }
+
+  const res = await resolveSources(".", options);
+
+  return res.configFile;
```

Comment:
I feel we could make function more generic like `resolePaths` or `resolveConfig` returning an object either full res or a subset, `cwd` and `configFile` are particulary useful. If we also resolve rc config, it can be useful for automation scripts to know where to edit/append it.